src/mono/wasm/runtime/binding_support.js

```diff
@@ -1576,7 +1679,7 @@ var BindingSupportLib = {
 			return BINDING.js_string_to_mono_string ("Invalid JS object handle '" + js_handle + "'");
 		}
 
-		var js_name = BINDING.conv_string (method_name);
+		var js_name = BINDING.conv_string (method_name, false);
```

Comment:
Method name is extremely likely to be interned here.  We should either check here or add a flag to Interop/Runtime::InvokeJSWithArgs

Reply:
I'm open to just interning it, this was an attempt to preserve behavior. I can't think of many cases where you would not want to intern the name of the method you're calling on a js object

Reply:
yeah, the paticular case I want to make sure we handle is stuff like
```csharp
        public int BytesPerElement => (int)GetObjectProperty("BYTES_PER_ELEMENT");
        public string Name => (string)GetObjectProperty("name");
        public int ByteLength => (int)GetObjectProperty("byteLength");
        public ArrayBuffer Buffer => (ArrayBuffer)GetObjectProperty("buffer");

        public void Fill(U value) => Invoke("fill", value);
        public void Fill(U value, int start) => Invoke("fill", value, start);
        public void Fill(U value, int start, int end) => Invoke("fill", value, start, end);

        public void Set(Array array) => Invoke("set", array);
        public void Set(Array array, int offset) => Invoke("set", array, offset);
        public void Set(ITypedArray typedArray) => Invoke("set", typedArray);
        public void Set(ITypedArray typedArray, int offset) => Invoke("set", typedArray, offset);

        public T Slice() => (T)Invoke("slice");
        public T Slice(int begin) => (T)Invoke("slice", begin);
        public T Slice(int begin, int end) => (T)Invoke("slice", begin, end);

        public T SubArray() => (T)Invoke("subarray");
        public T SubArray(int begin) => (T)Invoke("subarray", begin);
        public T SubArray(int begin, int end) => (T)Invoke("subarray", begin, end);
```
In the managed bindings

Reply:
If those are passed as mono_obj (managed ptr) it should be easy to make that work, hopefully they're not being passed as C strings

Reply:
They are, and they go through this and the other paths (properties, global object, new, etc..) my first thought is something like https://gist.github.com/lewing/0cc29d338d8cc1f2239f31bb1f36360b which works but doesn't exactly make the code cleaner.